### DISCUSSION TITLE
Data is lagging on some android phones.

### DISCUSSION THREAD

**Discussion by sbguzel** [PhilipsHue/flutter_reactive_ble#559] [Q&A]
I am working on an application that includes bluetooth communication. I take the accelerometer and gyro data from extarnal device to the phone and move the 3D model. While it works fine on IOS devices, some Android phones have data lag. Data is not lost, but act like buffered. Because it isn’t continues my 3D model lags. My model is following the device 1 second behind. IOS devices and very few android devices that I tested has no lag, working perfectly. I am using notify by the way. Am i missing something, or Is there any way to fix this problem.

Do you guys have any succession about that,  I would be very grateful if someone could give information about the problem.

Not: The device has ESP Wroom 32E. ESP send IMU data with 50Hz, GPS data with 20 Hz. IMU data payload is 19 byte, GPS payload is 87 byte.

**Comment by ahzf** [comment]
Do you use the ESP-IDF? Which version and which BLE framework?

**Reply by sbguzel** [reply]
Hi Achim Friedland, thank you for your comment. I am using https://www.digikey.com/en/products/detail/espressif-systems/ESP32-WROOM-32E-N4/11613125 with Arduino IDE. 

**BLE Libraries that i use are;**
#include <BLEDevice.h>
#include <BLEServer.h>
#include <BLEUtils.h>
#include <BLE2902.h>

**Notify Code :**
String str =  "DATA THAT I SEND";
char sendBuffer[100];
str.toCharArray(sendBuffer, str.length() + 1);
pTxCharacteristic->setValue((uint8_t*)sendBuffer, str.length() + 1);
pTxCharacteristic->notify();
